File: frontend/src/lib/preferences.ts
/**
 * Human labels and ordered option lists for the V1.2 personalisation
 * vocabularies.
 *
 * The canonical values live in the backend (`app/domain/preferences.py`); these
 * are the `{ value, label }` pairs onboarding and Settings render, plus label
 * maps for read-only display. Kept apart from `enums.ts` only because these
 * describe *who the user is and what they want to track* — personalisation that
 * shapes the UI — rather than the enums a data-entry form must offer.
 *
 * None of these ever reaches the analysis engine; they change which cards lead
 * and which fields a check-in surfaces first, nothing more.
 */

import type {
  FocusArea,
  HabitField,
  HouseholdContext,
  IncomePattern,
  LifeStage,
  WorkContextPref,
} from '../api/types';
import type { Option } from './enums';

export const LIFE_STAGE_OPTIONS: Option<LifeStage>[] = [
  { value: 'STUDENT', label: 'Student' },
  { value: 'EARLY_CAREER', label: 'Working professional' },
  { value: 'ESTABLISHED', label: 'Established / senior' },
  { value: 'FAMILY', label: 'Family / household manager' },
];

export const INCOME_PATTERN_OPTIONS: Option<IncomePattern>[] = [
  { value: 'SALARIED_FIXED', label: 'Fixed monthly salary' },
  { value: 'SALARIED_VARIABLE', label: 'Salary + variable pay' },
  { value: 'SELF_EMPLOYED', label: 'Self-employed / business' },
  { value: 'IRREGULAR', label: 'Irregular / allowance' },
];

export const WORK_CONTEXT_OPTIONS: Option<WorkContextPref>[] = [
  { value: 'OFFICE', label: 'Office' },
  { value: 'REMOTE', label: 'Remote' },
  { value: 'HYBRID', label: 'Hybrid' },
  { value: 'FIELD', label: 'On the field' },
];

export const HOUSEHOLD_CONTEXT_OPTIONS: Option<HouseholdContext>[] = [
  { value: 'LIVING_ALONE', label: 'Living alone' },
  { value: 'WITH_FAMILY', label: 'With family' },
  { value: 'WITH_PARTNER', label: 'With a partner' },
  { value: 'SHARED', label: 'Shared / flatmates' },
];

export const FOCUS_AREA_OPTIONS: Option<FocusArea>[] = [
  { value: 'UNDERSTAND_SPENDING', label: 'Understand where my money goes' },
  { value: 'BUILD_HEALTHY_HABITS', label: 'See how habits relate to spending' },
  { value: 'REDUCE_STRESS_SPENDING', label: 'Notice stress-driven spending' },
  { value: 'SAVE_MORE', label: 'Spend more deliberately' },
];

/** The six trackable habits, labelled. Values are the check-in field names. */
export const HABIT_OPTIONS: Option<HabitField>[] = [
  { value: 'sleep_minutes', label: 'Sleep' },
  { value: 'exercise', label: 'Exercise' },
  { value: 'home_cooked_meals', label: 'Home-cooked meals' },
  { value: 'stress_level', label: 'Stress' },
  { value: 'alcohol', label: 'Alcohol' },
  { value: 'work_mode', label: 'Work mode' },
];

function labelMap<T extends string>(options: Option<T>[]): Record<string, string> {
  return Object.fromEntries(options.map((option) => [option.value, option.label]));
}

export const LIFE_STAGE_LABELS = labelMap(LIFE_STAGE_OPTIONS);
export const INCOME_PATTERN_LABELS = labelMap(INCOME_PATTERN_OPTIONS);
export const WORK_CONTEXT_LABELS = labelMap(WORK_CONTEXT_OPTIONS);
export const HOUSEHOLD_CONTEXT_LABELS = labelMap(HOUSEHOLD_CONTEXT_OPTIONS);
export const FOCUS_AREA_LABELS = labelMap(FOCUS_AREA_OPTIONS);
export const HABIT_LABELS = labelMap(HABIT_OPTIONS);

File: domain/preferences.py
"""Closed vocabularies for onboarding and personalisation (V1.2, M5).

These describe *who the user is* and *what they want to pay attention to*. They
exist to **shape the UI** — which categories a check-in surfaces first, which
insight themes lead the dashboard, how onboarding introduces the product — and
for **nothing else**.

    A preference here NEVER enters the analysis engine.

That is a hard boundary, and it is structural rather than remembered: the engine
receives a plain dataset built from expenses, check-ins and life events
(``app/services/analysis_service.py``), never the ``User`` row, so a
personalisation field has no path into a statistical threshold, a gate, or a
hypothesis. The five gates (ADR-007) are the same for every user regardless of
what they selected here. ``tests/test_profile.py`` asserts the engine's source
never even mentions these fields.

Why closed sets, like the rest of the domain vocabulary (``enums.py``): a
personalisation field drives branching in the UI, and a bounded set keeps that
branching enumerable and testable. Adding a member is a schema decision, not a
free-text user action. These are deliberately coarse — enough to personalise,
not so fine that they become a survey.
"""

from __future__ import annotations

from enum import Enum


class LifeStage(str, Enum):
    """Roughly where the user is in life. Personalises copy and defaults only."""

    STUDENT = "STUDENT"
    EARLY_CAREER = "EARLY_CAREER"
    ESTABLISHED = "ESTABLISHED"
    FAMILY = "FAMILY"


class IncomePattern(str, Enum):
    """How income tends to arrive. Used to frame spending, never to predict it."""

    SALARIED_FIXED = "SALARIED_FIXED"
    SALARIED_VARIABLE = "SALARIED_VARIABLE"
    SELF_EMPLOYED = "SELF_EMPLOYED"
    IRREGULAR = "IRREGULAR"


class WorkContext(str, Enum):
    """The user's usual working arrangement.

    Distinct from :class:`app.domain.enums.WorkMode`, which is a *daily* fact
    recorded on a check-in. This is a stable preference set once at onboarding.
    """

    OFFICE = "OFFICE"
    REMOTE = "REMOTE"
    HYBRID = "HYBRID"
    FIELD = "FIELD"


class HouseholdContext(str, Enum):
    """Who the user shares expenses and daily life with."""

    LIVING_ALONE = "LIVING_ALONE"
    WITH_FAMILY = "WITH_FAMILY"
    WITH_PARTNER = "WITH_PARTNER"
    SHARED = "SHARED"


class FocusArea(str, Enum):
    """What the user says they want the product to help with.

    Coarse themes, not categories or habits — those are chosen separately
    (``tracked_categories`` / ``tracked_habits``). A focus area only changes
    which cards lead and which language onboarding uses.
    """

    UNDERSTAND_SPENDING = "UNDERSTAND_SPENDING"
    BUILD_HEALTHY_HABITS = "BUILD_HEALTHY_HABITS"
    REDUCE_STRESS_SPENDING = "REDUCE_STRESS_SPENDING"
    SAVE_MORE = "SAVE_MORE"
